src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -276,6 +280,26 @@ private boolean verify(
         kubectl, manifestObjects, consoleLogger, verifyTimeoutInMinutes);
   }
 
+  /**
+   * Get access token for service account with this credentialsId.
+   *
+   * @param credentialsId The service account credential's id.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws IOException If an error occurred fetching the access token.
+   */
+  public static String getAccessToken(String credentialsId) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+
+    ClientFactory clientFactory =
```

Comment:
The client factory creation is duplicated from getContainerClient method. I suggest re-factoring the getContainerClient method to just returning a client factory and removing the duplication here.

Reply:
I see what you're saying, but with the comment below on isolating the credentials-related functionalities, does it make sense to just use the credentials-related util functions for fetching the access token, thus I won't need a ClientFactory for the access token.

Reply:
Moved logic into CredentialsUtil class.